packages/backend-common/src/reading/AwsCodeCommitUrlReader.ts

```diff
@@ -0,0 +1,393 @@
+    integration: AwsCodeCommitIntegration,
+  ): Promise<CodeCommitClient> {
+    const credentials = await AwsCodeCommitUrlReader.buildCredentials(
+      credsManager,
+      region,
+      integration,
+    );
+
+    const codeCommit = new CodeCommitClient({
+      region: region,
+      credentials: credentials,
+    });
+    return codeCommit;
+  }
+
+  async readUrl(
+    url: string,
+    options?: ReadUrlOptions,
+  ): Promise<ReadUrlResponse> {
+    // etag and lastModifiedAfter are not supported by the CodeCommit API
+    try {
+      const { path, repositoryName, region, commitSpecifier } = parseUrl(
+        url,
+        true,
+      );
+      const codeCommitClient = await this.buildCodeCommitClient(
+        this.credsManager,
+        region,
+        this.integration,
+      );
+      const abortController = new AbortController();
+
+      const input: GetFileCommandInput = {
+        repositoryName: repositoryName,
+        commitSpecifier: commitSpecifier,
+        filePath: path,
+      };
+
+      options?.signal?.addEventListener('abort', () => abortController.abort());
+      const getObjectCommand = new GetFileCommand(input);
+      const response: GetFileCommandOutput = await codeCommitClient.send(
+        getObjectCommand,
+        {
+          abortSignal: abortController.signal,
+        },
+      );
+
+      if (options?.etag && options.etag === response.commitId) {
+        throw new NotModifiedError();
+      }
+
+      return ReadUrlResponseFactory.fromReadable(
+        Readable.from([response?.fileContent] || []),
+        {
+          etag: response.commitId,
+        },
+      );
+    } catch (e) {
+      if (e.$metadata && e.$metadata.httpStatusCode === 304) {
+        throw new NotModifiedError();
+      }
+      if (e.name && e.name === 'NotModifiedError') {
+        throw new NotModifiedError();
+      }
+
+      throw new ForwardedError('Could not retrieve file from CodeCommit', e);
+    }
+  }
+
+  async readTreePath(
+    codeCommitClient: CodeCommitClient,
+    abortSignal: any,
+    path: string,
+    repositoryName: string,
+    commitSpecifier?: string,
+    etag?: string,
+  ): Promise<string[]> {
+    const getFolderCommand = new GetFolderCommand({
+      folderPath: path,
+      repositoryName: repositoryName,
+      commitSpecifier: commitSpecifier,
+    });
+    const response = await codeCommitClient.send(getFolderCommand, {
+      abortSignal: abortSignal,
+    });
+
+    if (etag && etag === response.commitId) {
+      throw new NotModifiedError();
+    }
+
+    const output: string[] = [];
+    if (response.files) {
+      response.files.forEach(file => {
+        if (file.absolutePath) {
+          output.push(file.absolutePath);
+        }
+      });
+    }
+    if (!response.subFolders) {
+      return output;
+    }
+
+    for (const subFolder of response.subFolders) {
+      if (subFolder.absolutePath) {
+        output.push(
+          ...(await this.readTreePath(
```

Comment:
You probably want to forward `response.commitId` for all recursive requests, to make sure you don't hit a race were someone pushes a new commit in the middle of these requests